packages/core/src/App/Containers/AccountSignupModal/account-signup-modal.jsx

```diff
@@ -101,89 +94,72 @@ const AccountSignup = ({ enableApp, isModalVisible, clients_country, onSignup, r
                         touched,
                     }) => (
                         <Form>
-                            {!selected_residence || !is_same_citizenship_modal ? (
-                                <ResidenceForm
-                                    header_text={localize('Thanks for verifying your email')}
-                                    class_prefix='account-signup'
+                            {(!selected_residence && !selected_citizenship) || !is_pasword_modal ? (
```

Comment:
Why do you still need `selected_residence` and `selected_citizenship`?

Reply:
These checks were carried from the previous code. It checks whether the value for both of the field is present, if not it will show the modal. I do understand why it may be redundant because we already have `!is_pasword_modal` which is set to `false` by default.

Reply:
I don't think we need any of the flag since we have grouped both residency and citizenship modals.
help to remove

Reply:
Will do 😃 👍🏼